benchmarks/cholesky.bril

```diff
@@ -0,0 +1,335 @@
+  a13: float = const 83;
+  a20: float = const 67;
+  a21: float = const 92;
+  a22: float = const 56;
+  a23: float = const 92;
+  a30: float = const 67;
+  a31: float = const 826;
+  a32: float = const 38;
+  a33: float = const 43;
+
+  size: int = const 16;
+  return_ptr: ptr<float> = alloc size;
+  arr: ptr<float> = id return_ptr;
+  one: int = const 1;
+  store arr a00;
+  arr: ptr<float> = ptradd arr one;
+  store arr a01;
+  arr: ptr<float> = ptradd arr one;
+  store arr a02;
+  arr: ptr<float> = ptradd arr one;
+  store arr a03;
+  arr: ptr<float> = ptradd arr one;
+  store arr a10;
+  arr: ptr<float> = ptradd arr one;
+  store arr a11;
+  arr: ptr<float> = ptradd arr one;
+  store arr a12;
+  arr: ptr<float> = ptradd arr one;
+  store arr a13;
+  arr: ptr<float> = ptradd arr one;
+  store arr a20;
+  arr: ptr<float> = ptradd arr one;
+  store arr a21;
+  arr: ptr<float> = ptradd arr one;
+  store arr a22;
+  arr: ptr<float> = ptradd arr one;
+  store arr a23;
+  arr: ptr<float> = ptradd arr one;
+  store arr a30;
+  arr: ptr<float> = ptradd arr one;
+  store arr a31;
+  arr: ptr<float> = ptradd arr one;
+  store arr a32;
+  arr: ptr<float> = ptradd arr one;
+  store arr a33;
+
+  ret return_ptr;
+}
+
+# Fill array with zeros
+@zeros(sqsize: int) : ptr<float> {
+  arr: ptr<float> = alloc sqsize;
+  i: int = const 0;
+  one: int = const 1;
+  zero: float = const 0;
+.loop:
+  cond: bool = lt i sqsize;
+  br cond .body .done;
+.body:
+  loc: ptr<float> = ptradd arr i;
+  store loc zero;
+.loop_end:
+  i: int = add i one;
+  jmp .loop;
+.done:
+  ret arr;
+}
+
+# Prints an array
+@printarray(size: int, arr: ptr<float>) {
+  i: int = const 0;
+  one: int = const 1;
+.loop:
+  cond: bool = lt i size;
+  br cond .body .done;
+.body:
+  loc: ptr<float> = ptradd arr i;
+  val: float = load loc;
+  print val;
+.loop_end:
+  i: int = add i one;
+  jmp .loop;
+.done:
+  ret;
+}
+
+# Naive matrix muliplication algorithm
+# `arr1`, `arr2`, and `dest` are `size` x `size` matrices
+# `dest` will contain `arr1` x `arr2`
+@matmul(size: int, arr1: ptr<float>, arr2: ptr<float>, dest: ptr<float>) {
+  one: int = const 1;
+  row: int = const 0;
+.row.loop:
+  cond: bool = lt row size;
+  br cond .row.body .row.done;
+.row.body:
+  col: int = const 0;
+.col.loop:
+  cond: bool = lt col size;
+  br cond .col.body .col.done;
+.col.body:
+  sum: float = const 0.0;
+  i: int = const 0;
+.sum.loop:
+  cond: bool = lt i size;
+  br cond .sum.body .sum.done;
+.sum.body:
+  lidx: int = mul row size;
+  lidx: int = add lidx i;
+  ridx: int = mul i size;
+  ridx: int = add ridx col;
+  lvalloc: ptr<float> = ptradd arr1 lidx;
+  lval: float = load lvalloc;
+  rvalloc: ptr<float> = ptradd arr2 ridx;
+  rval: float = load rvalloc;
+  prod: float = fmul lval rval;
+  sum: float = fadd sum prod;
+.sum.loop_end:
+  i: int = add i one;
+  jmp .sum.loop;
+.sum.done:
+  idx: int = mul row size;
+  idx: int = add idx col;
+  loc: ptr<int> = ptradd dest idx;
```

Comment:
I think you meant `ptr<float>`

Reply:
Thanks, this issue is resolved by this PR: https://github.com/sampsyo/bril/pull/158, please review.